homework/tests/calculatortest.spec.js

```diff
@@ -0,0 +1,338 @@
+        await page.locator('#number1Field').type('2.2');
+        await page.locator('#number2Field').type('1.1');
+        await page.selectOption('#selectOperationDropdown', {label: 'Concatenate'});
+        await page.locator('#calculateButton').click();
+    
+        await expect(page.locator('#numberAnswerField')).toHaveValue('2.21.1');
+      });
+    });
+  });
+
+  data.forEach(version => {
+    test.describe(version + ': Integer checkbox', () => {
+      test('Integer checkbox not visible with non numeric values when concetenating', async ({ page }) => {
+        await page.goto('https://testsheepnz.github.io/BasicCalculator');
+        await page.selectOption('#selectBuild', { label: version});
+        await page.locator('#number1Field').type('i');
+        await page.locator('#number2Field').type('j');
+        await page.selectOption('#selectOperationDropdown', {label: 'Concatenate'});
+
+        await expect(page.locator('#integerSelect')).toBeHidden();        
+    
+        });
+    });
+  });
+
+  data.forEach(version => {
+    test.describe(version + ': Editable integer checkbox', () => {
+      test('Checking ability to edit integer checkbox', async ({ page }) => {
+        await page.goto('https://testsheepnz.github.io/BasicCalculator');
+        await page.selectOption('#selectBuild', { label: version});
+        await page.locator('#number1Field').type('6.2');
+        await page.locator('#number2Field').type('2.2');
+        await page.selectOption('#selectOperationDropdown', {label: 'Add'});
+        await page.locator('#calculateButton').click(); 
+    
+        await expect(page.locator('#integerSelect')).toBeEditable();
+      });
+    });
+  });
+ 
+  data.forEach(version => {
+    test.describe(version + ': Calculate button', () => {
+      test('Checking if Calculate button is visible', async ({ page }) => {
+        await page.goto('https://testsheepnz.github.io/BasicCalculator');
+        await page.selectOption('#selectBuild', { label: version});
+
+        await expect(page.locator('#calculateButton')).toBeVisible();       
+      });
+    });
+  });
+
+  data.forEach(version => {
+    test.describe(version + ': Build button', () => {
+      test('Checking if Build button is visible', async ({ page }) => {
+        await page.goto('https://testsheepnz.github.io/BasicCalculator');
+        await page.selectOption('#selectBuild', { label: version});
+
+        await expect(page.locator('#selectBuild')).toBeVisible();
+      });
+    });
+  });
+
+  data.forEach(version => {
+    test.describe(version + ': First number input', () => {
+      test('Checking if first number input is visible', async ({ page }) => {
+        await page.goto('https://testsheepnz.github.io/BasicCalculator');
+        await page.selectOption('#selectBuild', { label: version});
+       
+        await expect(page.locator('#number1Field')).toBeVisible();
+      });
+    });
+  });
+
```

Comment:
Those tests/checks could be removed as separate tests and other tests could be updated as in comment above:
![image](https://user-images.githubusercontent.com/73665598/180790469-d886783f-2c9b-498d-9437-1f0d5a6efc05.png)